lib/wit/gitrepo.py

```diff
@@ -242,14 +246,91 @@ def is_ancestor(self, ancestor, current=None):
         return proc.returncode == 0
 
     def repo_entries_from_commit(self, revision) -> List[RepoEntry]:
-        where = "{}:{}".format(revision, GitRepo.PKG_DEPENDENCY_FILE)
-        proc = self._git_command("show", where)
+        manifest_entries = self._read_manifest_from_commit(revision)
+        if len(manifest_entries) == 0:
+            manifest_entries = self._read_submodules_from_commit(revision)
+        return manifest_entries
+
+    def _read_manifest_from_commit(self, revision) -> List[RepoEntry]:
+        proc = self._git_command("show", "{}:{}".format(revision, GitRepo.PKG_DEPENDENCY_FILE))
         if proc.returncode:
-            log.debug("No dependency file found in repo [{}:{}]".format(revision,
+            log.debug("No wit dependency file found in repo [{}:{}]".format(revision,
                       self.path))
             return []
         return RepoEntries.parse(proc.stdout, Path(GitRepo.PKG_DEPENDENCY_FILE), revision)
 
+    def _read_submodules_from_commit(self, revision) -> List[RepoEntry]:
+        proc = self._git_command("show", "{}:{}".format(revision, GitRepo.SUBMODULE_FILE))
+        if proc.returncode:
+            log.debug("No .gitmodule file found in repo [{}:{}]".format(revision, self.path))
+            return []
+
+        log.debug("{}:{} does not have wit-manifest.json, "
+                  "reading dependencies from git submodules instead"
```

Comment:
```suggestion
                  "reading dependencies from .gitmodules instead"
```

Nit: I think explicitly naming the file you will read from is probably clearer than saying "git submodules", especially given that the debug-level log messages are probably only meant for wit maintainers to use.